src/nanotron/models/llama.py

```diff
@@ -322,8 +335,22 @@ def __init__(
             end=config.max_position_embeddings,
         )
 
-        # NOTE: Only supported for training (TODO(fmom): position_ids not supported yet)
-        self.flash_rotary_embedding = FlashRotaryEmbedding(dim=self.d_qk, interleaved=True)
+        if not DISABLE_FLASH_ATTENTION:
+            from flash_attn.layers.rotary import RotaryEmbedding as FlashRotaryEmbedding
+
+            self.flash_rotary_embedding = FlashRotaryEmbedding(dim=self.d_qk, interleaved=True)
+            # NOTE: Only supported for training (TODO(fmom): position_ids not supported yet)
+            # forward( self, qkv: torch.Tensor, kv)
+            self.attention = CoreAttention(
+                config,
+                parallel_config=parallel_config,
+                layer_idx=layer_idx,
+            )
+        else:
+            self.flash_rotary_embedding = RotaryEmbedding(
```

Comment:
why don't we use CoreAttention if DISABLE_FLASH_ATTENTION is False?

Reply:
Because it relies on the flash attention library. I want to disable all the function implementation from flash attention library.